lptree.pas

```diff
@@ -1662,22 +1662,49 @@ procedure TLapeTree_Invoke.setRealIdent(Node: TLapeTree_ExprBase);
 end;
 
 function TLapeTree_Invoke.getRealIdent(ExpectType: TLapeType): TLapeTree_ExprBase;
+
+  procedure CastOpenArrays(Method: TLapeType_Method);
+  var
+    i: Int32;
+  begin
+    if (FParams.Count <> Method.Params.Count) then
+      Exit;
+
+    for i := 0 to FParams.Count - 1 do
+      if FParams[i] is TLapeTree_OpenArray then
+        FParams[i] := FParams[i].setExpectedType(Method.Params[i].VarType) as TLapeTree_ExprBase;
```

Comment:
`setExpectedType` folds constants, which means it might set the type irreversibly. You might be able to work around it by just setting `ToType` and check `canCast` to check early if this overloaded function is viable.